list.cpp

```diff
@@ -0,0 +1,248 @@
+#include <iostream>
+using namespace std;
+
+struct Element
+{
+  int information;
+  Element * previous;
+  Element * next;   
+};
+
+Element * createElement()
+{
+  Element * el = new Element;
+  el->information = 0;
+  el->previous = 0;
+  el->next = 0;  
+
+  return el;
+} 
+
+Element * getPrevious(Element * el)
+{
+  return el -> previous;
+}
+
+Element * getNext(Element * el)
+{
+  return el -> next;
+}
+
+void insertAfter(Element * newEl, Element * oldEl)
+{
+  newEl->previous = oldEl;
+  newEl->next = oldEl->next;
+  oldEl->next = newEl;
+
+  if (newEl->next != 0) 
+    (newEl->next)->previous = newEl;
+}
+
+Element * insertBefore(Element * newEl, Element * oldEl, Element * head)
+{
+  newEl->previous = oldEl->previous;
+  oldEl->previous = newEl;
+  newEl->next = oldEl;
+
+  if (newEl->previous != 0)
+    (newEl->previous)->next = newEl;
+  else
+    head = newEl;
+
+  return head;
+}
+
+Element * deleteElement(Element * el, Element * head)
```

Comment:
Why this function returns head?
And it's better use not head but list as argument.